### PR TITLE
Fluent udf

### PR DESCRIPTION
**Pull request by BrandonHaynes**
UDF support in `myria-python`, including UDF registration via a decorator (`@myria_function`) and extension methods (`relation.select(lambda t: t.foo > 10).my_udf().store('result')`).

This PR still currently pulls from an explicit RACO branch, so shouldn't be merged yet.  But it can be reviewed :)

### CODE REVIEW

**Comment by BrandonHaynes** [comment]
@parmitam this PR is ready for review.

File: myria/fluent.py [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+        udf_name = name or f.__name__
+
+        setattr(
+            MyriaFluentQuery,
+            udf_name,
+            lambda self: self.select(**{udf_name: f,
+                                        'types': {udf_name: output_type}}))
+
+    return decorator
+
```
Comment (parmitam) [inline myria/fluent.py]: We need support for specifying multiValued in decorator...

**Comment by BrandonHaynes** [comment]
@parmitam I've updated the UDF extension method and added a test to verify that the server receives the `multivalued` bit.

**Comment by BrandonHaynes** [comment]
@parmitam all open issues addressed

**Comment by parmitam** [comment]
squash and merge done, ready for merge to master

**Comment by BrandonHaynes** [comment]
Superseded by PR #64

**Pull request closed without merging**